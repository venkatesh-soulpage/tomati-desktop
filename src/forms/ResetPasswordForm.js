import React from "react";
//redux
import { connect } from "react-redux";
import { resetPassword } from "_actions/auth";
// React router
import { Link, useLocation } from "react-router-dom";
//react bootstrap

// local components
import PasswordTextField from "components/PasswordTextField";

import { Card, Button, Form } from "react-bootstrap";

function ResetPasswordForm(props) {
  const [values, setValues] = React.useState({
    token: "",
    password: "",
    re_password: "",
  });

  const [alert, setAlert] = React.useState({
    variant: "warning",
    message: "Password does not match",
    switch: false,
  });
  const useQuery = () => {
    return new URLSearchParams(useLocation().search);
  };
  const query = useQuery();
  function onFormSubmit(event) {
    event.preventDefault();
    console.log("hitted", values);
    const email = query.get("email");
    const token = query.get("token");
    var data = {
      email,
      token,
      password: values.password,
    };
    if (values.password === values.re_password) {
      console.log(data, "DATA");
    } else {
      props.dispatch(resetPassword(data));
    }
  }

  const handleChange = (name) => (event) => {
    setValues({ ...values, [name]: event.target.value });
  };

  return (
    <div className="min-height mt-5 pt-5">
      <div className="container">
        <div className="row h-100 justify-content-center mt-5">
          <div className="col-md-6 w-100 align-self-center">
            <h2 className="text-dark mb-0 text-center mb-4">Update Password</h2>
            <Card className="p-5 pb-3" style={{ borderRadius: "15px" }}>
              <Form>
                <Form.Group>
                  <Form.Label>New Password</Form.Label>
                  <PasswordTextField
                    name="password"
                    value={values.password}
                    onChange={handleChange("password")}
                    placeholder="New Password"
                    autoComplete="off"
                  />
                </Form.Group>
                <Form.Group>
                  <Form.Control
                    type="password"
                    value={values.re_password}
                    onChange={handleChange("re_password")}
                    placeholder="Confirm New Password"
                  />
                </Form.Group>
                <div className="text-right">
                  <Form.Group className="mt-3">
                    <Button
                      variant="danger"
                      className="rounded-pill"
                      onClick={onFormSubmit}
                    >
                      Update
                    </Button>{" "}
                    <Button
                      type="submit"
                      variant="light"
                      className="rounded-pill"
                    >
                      Cancel
                    </Button>
                  </Form.Group>
                </div>
                <div className="text-muted">
                  <Link className="btn btn-link" to="/login">
                    Login as a different user
                  </Link>
                </div>
              </Form>
            </Card>
          </div>
        </div>
      </div>
    </div>
  );
}

function mapStateToProps(state) {
  return { auth: state.auth };
}
export default connect(mapStateToProps)(ResetPasswordForm);
