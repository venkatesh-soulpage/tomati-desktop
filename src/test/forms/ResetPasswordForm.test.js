import React from "react";
import { render } from "@testing-library/react";
import ResetPasswordForm from "forms/ResetPasswordForm";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("ResetPasswordForm page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ResetPasswordForm />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
