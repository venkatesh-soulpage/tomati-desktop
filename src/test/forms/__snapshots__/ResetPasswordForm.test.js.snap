// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm page after entering values 1`] = `
<DocumentFragment>
  <div
    class="min-height mt-5 pt-5"
  >
    <div
      class="container"
    >
      <div
        class="row h-100 justify-content-center mt-5"
      >
        <div
          class="col-md-6 w-100 align-self-center"
        >
          <h2
            class="text-dark mb-0 text-center mb-4"
          >
            Update Password
          </h2>
          <div
            class="p-5 pb-3 br-15 card"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="New Password"
                    type="password"
                    value="Soul@123"
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      class="show-button"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="Confirm New Password"
                    type="password"
                    value="Soul@123"
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      class="show-button"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-right"
              >
                <div
                  class="mt-3 form-group"
                >
                  <button
                    class="rounded-pill btn btn-danger"
                    type="button"
                  >
                    Update
                  </button>
                   
                  <a
                    href="/"
                  >
                    <button
                      class="rounded-pill btn btn-light"
                      type="button"
                    >
                      Cancel
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="text-muted"
              >
                <a
                  class="red-link"
                  href="/"
                >
                  Login as a different user
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ResetPasswordForm page before entering values 1`] = `
<DocumentFragment>
  <div
    class="min-height mt-5 pt-5"
  >
    <div
      class="container"
    >
      <div
        class="row h-100 justify-content-center mt-5"
      >
        <div
          class="col-md-6 w-100 align-self-center"
        >
          <h2
            class="text-dark mb-0 text-center mb-4"
          >
            Update Password
          </h2>
          <div
            class="p-5 pb-3 br-15 card"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="New Password"
                    type="password"
                    value=""
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      class="show-button"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="Confirm New Password"
                    type="password"
                    value=""
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      class="show-button"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-right"
              >
                <div
                  class="mt-3 form-group"
                >
                  <button
                    class="rounded-pill btn btn-danger"
                    type="button"
                  >
                    Update
                  </button>
                   
                  <a
                    href="/"
                  >
                    <button
                      class="rounded-pill btn btn-light"
                      type="button"
                    >
                      Cancel
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="text-muted"
              >
                <a
                  class="red-link"
                  href="/"
                >
                  Login as a different user
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
